UI overhaul for Test Site

imahmud1 opened the pull request.

imahmud1 committed:
HTML and JS files

imahmud1 committed:
Cleared vscode folder

GotenXiao commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,697 @@
+    <div class="container">
+        <!-- Header Section -->
+        <div class="header">
+            <img src="https://lancache.net/img/lancache-fulllogo.svg" alt="Logo">
```
This image should probably be baked into the container, rather than referenced off the LanCache site.

GotenXiao replied:
```suggestion
            <img src="https://lancache.net/img/lancache-fulllogo.svg" alt="LanCache.Net">
```

GotenXiao commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,697 @@
+
+        .status-dot {
+            width: 12px;
+            height: 12px;
```
This is pretty small on most modern displays. A nice option would be to declare a CSS property in `:root` (e.g. `--status-dot-size`, defaulted to `1rem`) and reference that here:
```suggestion
            width: var(--status-dot-size);
            height: var(--status-dot-size);
```

GotenXiao commented on lancache-litmus/overlay/var/www/litmus/sites.js:
```diff
@@ -0,0 +1,236 @@
+		]
+	},
+};
+var urltext='';
```
Ideally this shouldn't be part of this PR; I've added #6 to enable the container starting without an internet connection, as part of which a default `sites.js` will be generated at build time.

GotenXiao commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,697 @@
+            color: white;
+            font-size: 10px;
+            font-weight: bold;
+        }
```
```suggestion
        .status-dot.success::after {
            content: '✓';
            display: block;
            height: 100%;
            color: white;
            font-size: calc(var(--status-dot-size) / 2);
            font-weight: bold;
            text-align: center;
            line-height: var(--status-dot-size);
        }
```
Ideally the `font-size` would scale properly with the `--status-dot-size`, but this gets a little tricky. Might be simpler to make that a variable next to` --status-dot-size`.

GotenXiao commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,697 @@
+        <!-- Header Section -->
+        <div class="header">
+            <img src="https://lancache.net/img/lancache-fulllogo.svg" alt="Logo">
+            <h1>LanCache Test</h1>
```
Given that the logo is now present, I'm not sure this header adds anything useful.

GotenXiao commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,697 @@
+            background: rgba(233, 29, 18, 0.1);
+            padding: 1.5rem;
+            border-radius: 12px;
+            margin-bottom: 1.5rem;
```
```suggestion
```
As this is the last element in `div.header`, this effectively nearly the padding at the bottom.

GotenXiao commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,697 @@
+
+        .header img {
+            max-height: 80px;
+            margin-bottom: 1.5rem;
```
```suggestion
```
As the logo image already contains padding, this is superfluous.

GotenXiao commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,697 @@
+            font-size: 1.3rem;
+            font-weight: 700;
+            color: var(--text-color);
+            text-transform: capitalize;
```
```suggestion
            font-family: monospace;
```
Many of the service collection names don't look substantially better with initial capitalisation.

GotenXiao commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,697 @@
+        .alert .close:hover {
+            opacity: 1;
+        }
+
```
```suggestion

        #cache-info {
            position: sticky;
            top: 1em;
            z-index: 5;
        }
```

This way, even if a user is scrolled down the page, when they select a CDN any failures will be visible.

nightah commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,697 @@
+            $("#providercount").html(providercount);
+            cdntemplate.hide();
+        });
+    </script>
```
```suggestion
  <script>
      const wildcardPrefix = "lancachetest.";

      function hideAlert() {
        document.getElementById('cache-info').style.display = 'none';
      }

      function displayAlert(title, content, type = 'warning') {
        const alert = document.getElementById('cache-info');
        alert.classList.remove('alert-warning', 'alert-danger', 'alert-success');
        alert.classList.add(`alert-${type}`);
        document.getElementById('cache-info-display-title').innerHTML = `<strong>${title}</strong><br>`;
        document.getElementById('cache-info-display').innerHTML = content;
        alert.style.display = '';
      }

      function checkDomain(cdn, domainname) {
        domainname = domainname.replace(/^\*\./, wildcardPrefix);
        return new Promise(resolve => {
          const xhr = new XMLHttpRequest();
          xhr.open('GET', `//${domainname}/lancache-heartbeat`);
          xhr.onload = () => {
            const ret = {
              host: (domainname.includes(wildcardPrefix))
                ? domainname.replace(wildcardPrefix, '*.') + ` (via ${domainname})`
                : domainname,
              status: xhr.status === 204 ? 1 : 0,
              cdn,
              respondingHost: xhr.status === 204 ? xhr.getResponseHeader('X-LanCache-Processed-By') : false
            };
            resolve(ret);
          };
          xhr.onerror = () => resolve({ host: domainname, status: 0, cdn, respondingHost: false });
          xhr.send();
        });
      }

      document.addEventListener('DOMContentLoaded', () => {
        hideAlert();

        const input = document.getElementById('inputTest');
        const btn = document.getElementById('inputTestButton');

        input.addEventListener('keydown', e => {
          if (e.key === 'Enter') btn.click();
        });

        btn.addEventListener('click', () => {
          const url = input.value.trim();
          checkDomain(false, url).then(response => {
            if (response.status === 1) {
              displayAlert('✅ Test Succeeded', `${url} resolves to a LanCache instance named ${response.respondingHost}`, 'success');
            } else {
              displayAlert('❌ Test Failed', `${url} did not resolve to a LanCache instance`, 'danger');
            }
          });
        });

        const cdnContainer = document.getElementById('cdns');
        const template = document.getElementById('cdntemplate');
        template.style.display = 'none';

        let domaincount = 0, providercount = 0;
        const distinctCaches = [];

        for (const cdn in cachedomains) {
          providercount++;
          const domains = cachedomains[cdn].domains;
          domaincount += domains.length;
          cachedomains[cdn].brokenDomains = [];

          const el = template.cloneNode(true);
          el.id = `cdn-${cdn}`;
          el.style.display = '';
          cdnContainer.appendChild(el);

          el.querySelector('#title').textContent = cdn;
          el.querySelector('#tagline').textContent = cachedomains[cdn].description;
          el.querySelector('#totalcdndomaincount').textContent = domains.length;
          el.querySelector('#logo-fallback').textContent = cdn.charAt(0).toUpperCase();

          el.addEventListener('click', () => {
            const broken = cachedomains[cdn].brokenDomains;
            if (broken.length) {
              displayAlert('The following domains did not resolve to a LanCache instance:', broken.join('<br>'));
            } else {
              hideAlert();
            }
          });

          const checks = domains.map(d => checkDomain(cdn, d));
          Promise.all(checks).then(responses => {
            let success = 0, fail = 0;
            responses.forEach(r => {
              if (r.status === 1) {
                success++;
                distinctCaches.includes(r.respondingHost) || distinctCaches.push(r.respondingHost);
              } else {
                fail++;
                cachedomains[r.cdn].brokenDomains.push(r.host);
              }
            });

            const state = success === 0
              ? ['status-danger','danger']
              : (fail > 0
                ? ['status-warning','warning']
                : ['status-success','success']);

            const [cardClass, dotClass] = state;
            el.querySelector('.cdn-card')
              .className = `cdn-card ${cardClass}`;
            const dot = el.querySelector('.status-dot');
            dot.className = `status-dot ${dotClass}`;
            const countEl = el.querySelector('#workingcdndomaincount');
            countEl.className = `working-count ${dotClass}`;
            countEl.textContent = success;
          });
        }

        document.getElementById('domaincount').textContent = domaincount;
        document.getElementById('providercount').textContent = providercount;
      });
    </script>
```

This is a Vanilla JS version of so you can drop the jQuery requirement.

nightah commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,697 @@
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>LanCache Host Checker</title>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.6.0/jquery.min.js"></script>
```
```suggestion
```

You can drop this if you change the script to what has been suggested further down.

nightah commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,697 @@
+            font-weight: 600;
+            color: var(--text-color);
+        }
+
```
```suggestion

        #urltext a {
            color: var(--primary-color);
            text-decoration: underline;
        }

        #urltext a:hover {
          color: #c41611;
        }

```

Better colours for the `urltext` from `sites.js`.

nightah commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,697 @@
+                        🚀 Learn more about Lancache
+                    </a>
+                </div>
+            </div>
+        </div>
```
```suggestion
            </div>
            <p id="urltext"></p>
        </div>
```

Reinsert `urltext` to for page.

nightah reviewed (changes requested):
A few other small changes too please.

nightah commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,697 @@
+            
+            <div class="warning-banner">
+                🦊 Test this website in the Mozilla Firefox browser if the site doesnt work at all
+            </div>
```
```suggestion
```

Not required as the test works in Chrome/Chromium without issues.

nightah commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,697 @@
+            opacity: 0.95;
+            margin-bottom: 1.5rem;
+        }
+
+        .warning-banner {
+            background: rgba(233, 29, 18, 0.15);
+            padding: 1rem 1.5rem;
+            border-radius: 12px;
+            margin-bottom: 1.5rem;
+            font-weight: 600;
+            border: 1px solid rgba(233, 29, 18, 0.3);
+        }
```
```suggestion
```

Not required as the test works in Chrome/Chromium without issues.

imahmud1 committed:
Removed sites.js refactored and formatted html

imahmud1 commented:
Hello hello,

I hope you’re all doing well. Apologies for the delay I’ve been tied up with other tasks. I’m happy to report that I’ve implemented all of the suggested changes, and I’m really pleased with how everything turned out.

A few notes:

- As someone who isn’t formally trained in web development, your feedback was invaluable.

- I’ve removed the separate sites.js file and consolidated the UI and script changes directly into the HTML.

- Everything has been tested and is working smoothly on my end.

Please let me know if there’s anything else you’d like me to adjust or improve.

Thank you!

nightah commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,712 @@
+      .header .subtitle {
+        font-size: 1.25rem;
+        opacity: 0.95;
+        margin-bottom: 1.5rem;
```
```suggestion
```

Adjusted for missing `urltext`.

nightah commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,712 @@
+        font-weight: 600;
+        color: var(--text-color);
+      }
+
```
```suggestion

      #urltext {
        margin-bottom: 1.5rem;
      }

```

nightah commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,712 @@
+      #urltext a {
+        color: var(--primary-color);
+        text-decoration: underline;
+      }
```
```suggestion
      }

```

nightah commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,712 @@
+				<p class="subtitle">
+					We are aware of <strong id="domaincount">0</strong> domains spanning
+					<strong id="providercount">0</strong> providers.
+				</p>
+
```
```suggestion
				</p>
				<p id="urltext"></p>

```

nightah commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,712 @@
+          });
+        }
+        document.getElementById("domaincount").textContent = domaincount;
+        document.getElementById("providercount").textContent = providercount;
```
```suggestion
        document.getElementById("providercount").textContent = providercount;
        document.getElementById("urltext").innerHTML = urltext
```

nightah reviewed (changes requested):
Looks mostly good to me, only a few small things you missed from what I can tell.

imahmud1 committed:
More feedback changes

imahmud1 commented:
Can you kindly check now?

imahmud1 commented:
Im not exactly sure, but could you double check this section if we need any UI changes here:
<img width="1751" height="184" alt="image" src="https://github.com/user-attachments/assets/6d92c975-4610-4155-b8f5-a568631cf335" />
I think theres a bit spacing issue?

nightah commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,713 @@
+
+      .dns-info {
+        background: rgba(233, 29, 18, 0.1);
+        padding: 1.5rem;
```
```suggestion
        padding: 1rem;
```

nightah commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,713 @@
+
+      #urltext {
+        margin-bottom: 1.5rem;
+      }
+
```
```suggestion
      }

      #urltext a {
        color: var(--primary-color);
        text-decoration: underline;
      }
      
```

nightah commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,713 @@
+      }
+
+      #urltext {
+        margin-bottom: 1.5rem;
```
```suggestion
        margin-bottom: 1rem;
```

nightah commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,713 @@
+				<p class="subtitle">
+					We are aware of <strong id="domaincount">0</strong> domains spanning
+					<strong id="providercount">0</strong> providers.
+				</p>
+        <p id="urltext"></p>
```
```suggestion
				</p>
				<p id="urltext"></p>
```

nightah commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,713 @@
+        <p id="urltext"></p>
+
+				<div class="dns-info">
+					<p style="margin-bottom: 1rem">
```
```suggestion
					<p>
```

nightah commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,713 @@
+						</a>
 					</div>
 				</div>
+				<p id="urltext"></p>
```
```suggestion
```

nightah commented on lancache-litmus/overlay/var/www/litmus/index.html:
```diff
@@ -1,208 +1,713 @@
-					<span id="cache-info-display">
-					</span>
 				</div>
+				<p id="urltext"></p>
```
```suggestion
```

nightah reviewed (changes requested):
I had a look at your spacing issue and while I don't think it was a huge problem I've made some minor adjustments, let me know what you think.

There were a few other small changes too.

imahmud1 committed:
Refactored and formatted

imahmud1 commented:
UI looks good to me. Im not sure how the formatting broke. Ive formatted again. And made all the changes.

imahmud1 commented:
> UI looks good to me. Im not sure how the formatting broke. Ive formatted again. And made all the changes.

Do let me know if i missed anything else.

nightah reviewed (approved):
LGTM!

I'll let @GotenXiao cast his eyes over this one last time for completeness.

GotenXiao reviewed (approved):
LGTM.
